python/pyarrow/tests/test_table.py

```diff
@@ -926,36 +926,46 @@ def test_recordbatch_to_tensor_uniform_type(typ):
             pa.array(arr3, type=pa.from_numpy_dtype(typ)),
         ], ["a", "b", "c"]
     )
-    result = batch.to_tensor()
 
+    result = batch.to_tensor(row_major=False)
     x = np.array([arr1, arr2, arr3], typ).transpose()
     expected = pa.Tensor.from_numpy(x)
+    check_tensors(result, expected, pa.from_numpy_dtype(typ), 27)
 
+    result = batch.to_tensor()
+    x = np.array([arr1, arr2, arr3], typ, order='F').transpose()
+    expected = pa.Tensor.from_numpy(x)
```

Comment:
```suggestion
    result = batch.to_tensor(row_major=False)
    x = np.column_stack([arr1, arr2, arr3]).astype(typ, order=F")
    expected = pa.Tensor.from_numpy(x)
    check_tensors(result, expected, pa.from_numpy_dtype(typ), 27)

    result = batch.to_tensor()
    x = np.column_stack([arr1, arr2, arr3]).astype(typ, order=C")
    expected = pa.Tensor.from_numpy(x)
```

This might be a clearer way to create the expected numpy arrays? Then you don't need to think about how the `np.array(..)` constructor will create the array (stack as rows), and then we don't have the `order='F').transpose()` to create row major data.

(and same for some occurrences below)